third-party/rspamd/src/main/java/org/apache/james/rspamd/task/FeedSpamToRSpamDTask.java

```diff
@@ -0,0 +1,389 @@
+            this.spamMessageCount = new AtomicLong();
+            this.reportedSpamMessageCount = new AtomicLong();
+            this.errorCount = new AtomicLong();
+            this.messagesPerSecond = runningOptions.messagesPerSecond;
+            this.period = runningOptions.period;
+            this.samplingProbability = runningOptions.samplingProbability;
+        }
+
+        public void incrementSpamMessageCount() {
+            spamMessageCount.incrementAndGet();
+        }
+
+        public void incrementReportedSpamMessageCount(int count) {
+            reportedSpamMessageCount.addAndGet(count);
+        }
+
+        public void incrementErrorCount() {
+            errorCount.incrementAndGet();
+        }
+
+        public Snapshot snapshot() {
+            return Snapshot.builder()
+                .spamMessageCount(spamMessageCount.get())
+                .reportedSpamMessageCount(reportedSpamMessageCount.get())
+                .errorCount(errorCount.get())
+                .messagesPerSecond(messagesPerSecond)
+                .period(period)
+                .samplingProbability(samplingProbability)
+                .build();
+        }
+    }
+
+    private final GetMailboxMessagesService messagesService;
+    private final RSpamDHttpClient rSpamDHttpClient;
+    private final RunningOptions runningOptions;
+    private final Context context;
+    private final Clock clock;
+
+    public FeedSpamToRSpamDTask(MailboxManager mailboxManager, UsersRepository usersRepository, RSpamDHttpClient rSpamDHttpClient,
+                                RunningOptions runningOptions, Clock clock) {
+        this.runningOptions = runningOptions;
+        this.messagesService = new GetMailboxMessagesService(mailboxManager, usersRepository);
+        this.rSpamDHttpClient = rSpamDHttpClient;
+        this.context = new Context(runningOptions);
+        this.clock = clock;
+    }
+
+    @Override
+    public Result run() {
+        Optional<Date> afterDate = runningOptions.period.map(periodInSecond -> Date.from(clock.instant().minusSeconds(periodInSecond)));
+        try {
+            return messagesService.getMailboxMessagesOfAllUser(SPAM_MAILBOX_NAME)
+                .doOnNext(messageResult -> context.incrementSpamMessageCount())
+                .filter(messageResult -> afterDate.map(date -> messageResult.getInternalDate().after(date)).orElse(true))
+                .filter(any -> randomBooleanWithProbability(runningOptions.samplingProbability))
+                .transform(ReactorUtils.<MessageResult, Task.Result>throttle()
+                    .elements(runningOptions.messagesPerSecond)
+                    .per(Duration.ofSeconds(1))
+                    .forOperation(messageResult1 -> Mono.just(messageResult1)
+                        .flatMap(Throwing.function(messageResult -> rSpamDHttpClient.reportAsSpam(messageResult.getHeaders().getInputStream(), messageResult.getBody().getInputStream())))
```

Comment:
```suggestion
                    .forOperation(messageResult1 -> messageResult -> rSpamDHttpClient.reportAsSpam(messageResult.getHeaders().getInputStream(), messageResult.getBody().getInputStream())))
```

Please simpify this.

Reply:
Switching to this, we would have to do a verbose try catch to use `.fullContent`. I put in some effort but did not find a better solution.

Reply:
Yes but we could wrap this using `Throwing.function` in the right place, right?